### PR TITLE
#162727478 Add search functionality

### PR DESCRIPTION
**Pull request by daniellamarr**
#### What does this PR do?
- This PR integrates search functionality for the app
#### Description of Task to be completed?
- A user should be able to search for articles based on categories
- A user should be able to search for articles based on keywords passed
- A user can search for other users
#### How should this be manually tested?
- Run `npm install`, then `npm start`
- Access the route `/api/v1/search/categories/:categoryId` to search arts based on the category id passed.
- Access the route `/api/v1/search/:keyword` to search arts based on the keyword passed.
- Access the route `/api/v1/search/users/:user` to search for available users on the platform.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#162727478](https://www.pivotaltracker.com/story/show/162727478)
#### Screenshots (if appropriate)
screen shot 2019-01-31 at 5 33 03 pm
screen shot 2019-01-31 at 5 33 13 pm
screen shot 2019-01-31 at 5 33 19 pm

#### Questions:
N/A

### CODE REVIEW

**Review by abejide001 — APPROVED** [review]
Looks good to me

**Review by Proception — APPROVED** [review]
Looks good

**Review by wombolo — APPROVED** [review]
Looks Good

File: src/controllers/SearchController.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+class SearchController {
+  /**
+   * @static
+   * @desc POST /api/v1/search/categories/:categoryId
+   * @param {object} req
+   * @param {object} res
+   * @memberof SearchController
+   * @returns {object} arts matching the specified category
+   */
+  static async searchByCategory(req, res) {
```
Comment (andela-dbamidele) [inline src/controllers/SearchController.js]: This is searching by the category id. Searching is mostly done by name or title

**Comment by andela-dbamidele** [comment]
Please look at the comment and fix the merge conflict @daniellamarr

File: src/controllers/SearchController.js (hunk 1 above)
Comment (daniellamarr) [reply src/controllers/SearchController.js]: Yes, so the idea is on the frontend the users get to see the category URL but on the backend, the id is being queried.

Comment (andela-dbamidele) [reply src/controllers/SearchController.js]: Then the function should be named differently. Something like `getArtsByCategory`.

**Pull request merged**